crates/saya/core/src/verifier/starknet.rs

```diff
@@ -1,75 +1,81 @@
-use std::time::Duration;
-
 use anyhow::Context;
 use dojo_world::migration::TxnConfig;
 use dojo_world::utils::TransactionExt;
+use itertools::Itertools;
 use starknet::accounts::{Account, Call, ConnectedAccount};
-use starknet::core::types::{Felt, TransactionExecutionStatus, TransactionStatus};
+use starknet::core::types::Felt;
 use starknet::core::utils::get_selector_from_name;
-use starknet::providers::Provider;
-use tokio::time::sleep;
+use starknet_crypto::poseidon_hash_many;
+use tracing::trace;
 
-use crate::dojo_os::get_starknet_account;
-use crate::StarknetAccountData;
+use super::utils::wait_for_sent_transaction;
+use crate::SayaStarknetAccount;
 
 pub async fn starknet_verify(
     fact_registry_address: Felt,
     serialized_proof: Vec<Felt>,
-    starknet_config: StarknetAccountData,
+    cairo_version: Felt,
+    account: &SayaStarknetAccount,
 ) -> anyhow::Result<(String, Felt)> {
+    if serialized_proof.len() > 2000 {
```

Comment:
isn't `4000` the limit?

Reply:
I exprienced that some transactions failed when greater than 3000, but can test further now that it's in a more mature state.